### ISSUE TITLE
Apex Golang shim doesn't update environment variables

### ISSUE THREAD

**Issue by DaveBlooman** [apex/apex#817]
So, this is a very specific issue around AWS generated env vars, specifically, the x-ray trace-id.  Xray is a service that generates a series env vars, trace ID and parent ID, but those env vars are not present initially when the lambda executes.  

I would imagine that because of the way that lambda caches containers, it wont bother creating the env vars at the initial runtime, however, when the handler is called, new env vars are generated and handed to the process.  This wont happen for any Go shim as the process is already invoked by the time the handler is called.  You can see this by turning on tracing and running the following code.

```js
console.log('start using-env xray=%s', process.env._X_AMZN_TRACE_ID)
exports.handle = function(e, ctx, cb) {
  console.log('start using-env xray=%s', process.env._X_AMZN_TRACE_ID)
    
  cb(null, {
    hello: 'bar',
    token_used: process.env._X_AMZN_TRACE_ID
  })
}

```

Perhaps the shim could be changed to feed the env vars in a different way, i'm not much of a nodejs guy, but im sure this is possible.

**Comment by tj** [comment]
Hmm, they really update env vars every request with the Node runtime? Seems like an odd use of env vars, especially when they have the context object. I haven't used x-ray with Lambda yet though (only some custom stuff)

**Comment by DaveBlooman** [comment]
Yeah, I understand why, is there something that can be done in the shim to refresh?

**Comment by tj** [comment]
Not that I know of, I think only the sub-proc can setenv() to alter its own, maybe some `/proc/` hack, but that's really messed up on AWS' part if they did that haha

**Comment by DaveBlooman** [comment]
Yeah, its a real weird thing AWS are going, i'm gong to get in touch with them, see what they say.  At the very least, it won't be possible to use the Apex shim for Go if you want to use X-Ray :(
